docs/getting-started/server-rendering.md

```diff
@@ -1,57 +1,179 @@
 # Server Rendering
 
-When using Material-UI with server rendering, we must use the same environment for the server and the client.
-This has two technical implications.
+The most common use case for server-side rendering is to handle the *initial render* when a user (or search engine crawler) first requests our app.
+When the server receives the request, it renders the required component(s) into an HTML string, and then sends it as a response to the client.
+From that point on, the client takes over rendering duties.
 
-## Autoprefixer
+## Material-UI on the Server
 
-First, Material-UI has to use the same user agent for the auto prefixer.
-On the client side, the default value is `navigator.userAgent`.
-But on the server side, the `navigator` is `undefined`. You need to provide it to Material-UI.
+Material-UI was designed from the ground-up with the constraint of rendering on the Server.
+But it's up to users to makes sure it's correctly integrated.
+We must provide to the page the needed style.
+It's important because you don't want the interface to flicker.
+Otherwise, the page will render with just the HTML, then wait for the CSS to be injected.
+To inject the style down to the client, we need to:
+- create a fresh, new `styleManager` and `theme` instance on every request;
+- render the React tree with the server-side API and the instance;
+- pull the CSS out of the `styleManager`;
+- and then pass the CSS along to the client.
```

Comment:
Up to you, but I would make this a numbered list, and begin each item with a capital, and end with each a full-stop. The content itself is good.